### PR TITLE
installer/pkg/validate: validate pull secret and license paths

### PR DESCRIPTION
**Pull request by squat**
This commit adds validation for the pull secret and license paths. These fields are often forgotten or incorrect. These changes add basic type validation for the files located at the provided paths, namely ensure that the pull secret file is valid JSON and ensure that the license is a valid JWT.

The vast majority of this changeset is unit tests.

Addresses: INST-1114

cc @enxebre @spangenberg

### CODE REVIEW

File: installer/pkg/config/validate.go [hunk 1]
```diff
@@ -206,14 +206,31 @@ func (c *Cluster) validateTNCS3Bucket() error {
 	return nil
 }
 
+func (c *Cluster) validateTectonicFiles() []error {
+	var errs []error
+	if err := validate.FileExists(c.PullSecretPath); err != nil {
```
Comment (enxebre) [inline installer/pkg/config/validate.go]: I think we could omit `FileExists` here as it's implicit in `JSONFile` which calls `ioutil.ReadFile` wdyt?

File: installer/pkg/config/validate.go [hunk 2]
```diff
@@ -206,14 +206,31 @@ func (c *Cluster) validateTNCS3Bucket() error {
...
+		errs = append(errs, err)
+	} else if err := validate.JSONFile(c.PullSecretPath); err != nil {
+		errs = append(errs, err)
+	}
+	if err := validate.FileExists(c.LicensePath); err != nil {
```
Comment (enxebre) [inline installer/pkg/config/validate.go]: I think we could omit `FileExists` here as it's implicit in `License` which calls `ioutil.ReadFile` wdyt?

File: installer/pkg/config/validate.go (hunk 1 above)
Comment (squat) [reply installer/pkg/config/validate.go]: yes sounds good

File: installer/pkg/config/validate.go (hunk 2 above)
Comment (squat) [reply installer/pkg/config/validate.go]: agreed

**Comment by squat** [comment]
need one more review. cc @alexsomesan

**Comment by squat** [comment]
had to rebase again because of new merge conflicts. we need to get this reviewed and merged asap.

cc @enxebre @alexsomesan

**Comment by squat** [comment]
thanks again, @enxebre

**Comment by squeed** [comment]
lgtm. quick! merge!

**Comment by squat** [comment]
@alexsomesan yes for sure. I'll follow up with that change :+1:

**Pull request merged**